static/app/components/deprecatedDropdownMenu.spec.tsx

```diff
@@ -0,0 +1,210 @@
+    userEvent.click(screen.getByTestId('outside-element'));
+
+    await waitForElementToBeRemoved(() => screen.queryByRole('listbox'));
+  });
+
+  it('closes dropdown when pressing escape', function () {
+    render(<DeprecatedDropdownImplementation />);
+    userEvent.click(screen.getByRole('button'));
+
+    userEvent.keyboard('{Escape}');
+    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
+  });
+
+  it('ignores "Escape" key if `closeOnEscape` is false', function () {
+    render(<DeprecatedDropdownImplementation closeOnEscape={false} />);
+    userEvent.click(screen.getByRole('button'));
+
+    userEvent.keyboard('{Escape}');
+    expect(screen.getByRole('listbox')).toBeInTheDocument();
+  });
+
+  it('keeps dropdown open when clicking on anything in menu with `keepMenuOpen` prop', function () {
+    render(<DeprecatedDropdownImplementation keepMenuOpen />);
+    userEvent.click(screen.getByRole('button'));
+    userEvent.click(screen.getByRole('listitem'));
+
+    expect(screen.getByRole('listbox')).toBeInTheDocument();
+  });
+
+  it('render prop getters all extend props and call original onClick handlers', function () {
+    const rootClick = jest.fn();
+    const actorClick = jest.fn();
+    const menuClick = jest.fn();
+
+    render(
+      <DeprecatedDropdownMenu keepMenuOpen>
+        {({getRootProps, getActorProps, getMenuProps, isOpen}) => (
+          <span
+            {...getRootProps({
+              onClick: rootClick,
+            })}
+            data-test-id="root"
+          >
+            <button
+              {...getActorProps({
+                onClick: actorClick,
+              })}
+              data-test-id="actor"
+            >
+              Open Dropdown
+            </button>
+            {isOpen && (
+              <ul
+                {...getMenuProps({
+                  onClick: menuClick,
+                })}
+                data-test-id="menu"
+              >
+                <li>Dropdown Menu Item 1</li>
+              </ul>
+            )}
+          </span>
+        )}
+      </DeprecatedDropdownMenu>
+    );
+
+    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
+
+    userEvent.click(screen.getByTestId('root'));
+    expect(rootClick).toHaveBeenCalled();
+
+    userEvent.click(screen.getByTestId('actor'));
+    expect(actorClick).toHaveBeenCalled();
+
+    userEvent.click(screen.getByTestId('menu'));
+    expect(menuClick).toHaveBeenCalled();
+
+    expect(screen.queryByRole('listbox')).toBeInTheDocument();
+  });
+
+  it('always rendered menus should attach document event listeners only when opened', function () {
+    const addSpy = jest.spyOn(document, 'addEventListener');
+    const removeSpy = jest.spyOn(document, 'removeEventListener');
+
+    render(
+      <DeprecatedDropdownMenu alwaysRenderMenu>
+        {({getRootProps, getActorProps, getMenuProps}) => (
+          <span
+            {...getRootProps({
+              className: 'root',
+            })}
```

Comment:
super tiny nit, but this can prob be one line
```suggestion
            {...getRootProps({className: 'root'})}
```